pilot/pkg/xds/proxy_dependencies.go

```diff
@@ -73,11 +73,13 @@ func checkProxyDependencies(proxy *model.Proxy, config model.ConfigKey, push *mo
 			}
 
 			// If config is ServiceEntry, name of the config is service's FQDN
-			svc, exist := push.ServiceIndex.HostnameAndNamespace[host.Name(config.Name)][config.Namespace]
-			if exist {
-				if !push.IsServiceVisible(svc, proxy.Metadata.Namespace) {
-					return false
-				}
+			hostname := host.Name(config.Name)
+			// gateways have default sidecar scopes
+			if proxy.SidecarScope.GetService(hostname) == nil &&
+				proxy.PrevSidecarScope.GetService(hostname) == nil {
```

Comment:
BTW I think we could expand this to other configs(VS, DR) ? just like what we do for sidecars